### PR TITLE
.gitignore: add *.bak, *.orig, files generated by VS, and reorganize .gitignore file

### PR DESCRIPTION
**Pull request by dilijev**
Added
*.bak
*.orig

Added many more types of files that are sometimes generated by VS.

Organized ignore filters into categories, and sorted lines within categories.

### CODE REVIEW

**Comment by pleath** [comment]
Yay.

File: .gitignore [hunk 1]
```diff
@@ -25,6 +25,7 @@ ComputedEnvironment.cmd
 packages/
 *.tlog/
 .*.swp
+*.orig
```
Comment (obastemur) [inline .gitignore]: Thanks for this. Could you also consider adding `*.bak` too ?

Comment (dilijev) [inline .gitignore]: Sure. What tools generate *.bak?

**Comment by dilijev** [comment]
As long as we're tweaking, here's my gitignore_global:

```
#ignore thumbnails created by windows
Thumbs.db
#Ignore files build by Visual Studio
*.obj
*.exe
*.pdb
*.user
*.aps
*.pch
*.vspscc
*_i.c
*_p.c
*.ncb
*.suo
*.tlb
*.tlh
*.bak
*.cache
*.ilk
*.log
*.dll
*.lib
*.sbr

.vscode

*~
~*
*.orig

#vim
.*.swp
.*.swo
```

File: .gitignore (hunk 1 above)
Comment (obastemur) [inline .gitignore]: http://manual.winmerge.org/Configuration.html [11.3]

**Comment by dilijev** [comment]
Added the above, organized ignore filters into categories, and sorted lines within categories.

**Comment by tcare** [comment]
Looks good :)

**Pull request merged**